Dockerfile

```
@@ -12,41 +14,52 @@ ARG USER_ID
 ARG USER_NAME
 ARG HOME_DIR
 ARG SOURCE_DIR
+ARG NPM_REGISTRY_URL
+ARG NODE_ENV
+
+ENV NODE_ENV=$NODE_ENV
 
 # Create the group (use a high ID to attempt to avoid conflits).
-RUN groupadd -g $USER_ID $USER_NAME
+RUN groupadd --non-unique -g $USER_ID $USER_NAME
 
 # Create the user (use a high ID to attempt to avoid conflits).
-RUN useradd -d $HOME_DIR -m -u $USER_ID -g $USER_ID $USER_NAME
-
-# Update the system.
-RUN apt-get update && apt-get upgrade -y
-
-# Install Nodejs and npm.
-RUN apt-get install -y nodejs
-RUN apt-get install -y npm
+RUN useradd --non-unique -d $HOME_DIR -m -u $USER_ID -g $USER_ID $USER_NAME
+
+# Install stable Nodejs and npm.
+RUN \
+  apt-get update \
+  && apt-get -y install nodejs npm \
```

This **must** have the `ip` program installed to work under linux.

```suggestion
  && apt-get -y install nodejs npm ip \
```

---

Added